src/utils/canister/actor.ts

```diff
@@ -15,20 +15,62 @@ function getHost() {
 
 const host = getHost();
 
-// Since we're using webpack-dev-server as part of create-react-app, we need to
-// add its port to our HttpAgent config as the host.
-const agent = new HttpAgent({ host });
-export const baseActor = Actor.createActor<_SERVICE>(CanCan_idl, {
-  agent,
-  canisterId: CanCan_canister_id,
-});
-
-export class CanCanActor {
-  actor: _SERVICE;
-  constructor(overrideActor?: _SERVICE) {
-    this.actor = overrideActor ?? baseActor;
+function createActor(identity?: Identity) {
+  const agent = new HttpAgent({ host, identity });
+  const actor = Actor.createActor<_SERVICE>(CanCan_idl, {
+    agent,
+    canisterId: CanCan_canister_id,
+  });
+  return { actor, agent };
+}
+
+/*
+ * Responsible for keeping track of the actor, whether the user has logged
+ * in again or not. A logged in user uses a different actor with their
+ * Identity, to ensure their Principal is passed to the backend.
+ */
+class ActorController {
+  _actor: _SERVICE;
+  _isAuthenticated: boolean = false;
+  isReady: boolean = false;
+
+  constructor() {
+    this._actor = this.initBaseActor();
+  }
+
+  initBaseActor() {
+    const { agent, actor } = createActor();
+    agent.fetchRootKey().then(() => (this.isReady = true));
```

Comment:
I believe it's not safe to do this when deployed on real ic, because someone could [MITM](https://owasp.org/www-community/attacks/Manipulator-in-the-middle_attack).
It's only safe when developing against an IC replica that the end-user 'trusts'. e.g. if in dev mode.